backend/fms_core/template_importer/row_handlers/sample_qc/sample.py

```diff
@@ -72,7 +76,9 @@ def process_row_inner(self, sample, sample_information, process_measurement, pro
                         self.errors['instrument'] = f'Invalid instrument {type}.'
 
                 # Validate required RIN for RNA
-                if sample_obj.derived_sample_not_pool.sample_kind.name == 'RNA' and process_measurement_properties['RIN']['value'] is None:
-                    self.errors['RIN'] = 'RIN has to be specified for RNA.'
+                # For pools we check all sample kinds
+                for derived_sample in sample_obj.derived_samples.all():
+                    if derived_sample.sample_kind.name == 'RNA' and process_measurement_properties['RIN']['value'] is None:
```

Comment:
I think this may not be the optimal way to deal with this. Users will receive multiple errors for a single pooled sample if the sample kind is RNA. First sample pools should not contain different sample_kind. I think we should enforce that (i have to double check in the current pool service if it is the case). In this case i think we should make a sample property that return a single sample kind for sample pools (they should all be the same) and it would return pool for pooled libraries (meaning we don't care too much at this point and we are not enforcing anything about those).
Here you should be able to ask sample_obj.sample_kind == "RNA" and ...: a single time for your sample and have a single error message.